docs/mp/testing-ng.adoc

```diff
@@ -42,20 +51,68 @@ A test can be annotated with `io.helidon.microprofile.tests.testng.HelidonTest`
+
+This will change the behavior as follows:
+
+- A new CDI container is created for each test method invocation
+- annotations to add config, beans and extension can be added for each method in addition to the class
+- you cannot inject fields or constructor parameters of the test class itself (as a single instance is shared by more containers)
+
+=== Usage - configuration
+In addition to the `@AddConfig` annotation, you can also use
+`@Configuration` to configure additional classpath properties config sources using `configSources`, and to
+mark that a custom configuration is desired.
+If `@Configuration(useExisting=true)`, the existing (or default) MicroProfile configuration would be used. In this case
+it is important to set property `mp.initializer.allow=true` in order CDI container to start, when used with
+`@HelidonTest`.
+You can set up config in `@BeforeAll` method and register it with `ConfigProviderResolver` using MP Config APIs, and declare
+`@Configuration(useExisting=true)`.
+Note that this is not compatible with repeatable tests that use method sources that access CDI, as we must delay the CDI
+startup to the test class instantiation (which is too late, as the method sources are already invoked by this time).
+
+*If you want to use method sources that use CDI with repeatable tests, please do not use `@Configuration(useExisting=true)`*
+
+NOTE: Test method parameters are currently not supported.
+
+== API
+
 The annotations described in this section are inherited (for the non-repeatable ones), and additive (for repeatable).
 So if you declare `@DisableDiscovery` on abstract class, all implementations will have discovery disabled, unless you
 annotate the implementation class with `@DisableDiscovery(false)`.
 If you declare `@AddBean` on both abstract class and implementation class, both beans will be added.
 
 In addition to this simplification, the following annotations are supported:
 
-- `io.helidon.microprofile.tests.testng.AddBean` - to add one or more beans to the container
-        (if not part of a bean archive, or when discovery is disabled)
-- `io.helidon.microprofile.tests.testng.AddExtension` - to add one or more CDI extensions to the container
-        (if not added through service loader, or when discovery is disabled)
-- `io.helidon.microprofile.tests.testng.AddConfig` - to add one or more configuration properties to MicroProfile config
-        without the need of creating a `microprofile-config.properties` file
-- `io.helidon.microprofile.tests.testng.DisableDiscovery` - to disable automated discovery of beans and extensions
+
+|===
+|Annotation | Usage
+
+|`@io.helidon.microprofile.tests.testng.AddBean`
+|to add one or more beans to the container (if not part of a bean archive, or when discovery is disabled)
+
+|`@io.helidon.microprofile.tests.testng.AddExtension`
+|to add one or more CDI extensions to the container (if not added through service loader, or when discovery is disabled)
```

Comment:
I would add `Use` as the first word to make this sentence readable.

Reply:
Done